src/impl/android/external_commit_helper.cpp

```diff
@@ -89,24 +89,34 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
     int ret = mkfifo(path.c_str(), 0600);
     if (ret == -1) {
         int err = errno;
-        if (err == ENOTSUP) {
+        if (err == ENOTSUP || err == EACCES) {
             // Filesystem doesn't support named pipes, so try putting it in tmp instead
             // Hash collisions are okay here because they just result in doing
             // extra work, as opposed to correctness problems
-            std::ostringstream ss;
-
-            std::string tmp_dir(getenv("TMPDIR"));
-            ss << tmp_dir;
-            if (tmp_dir.back() != '/')
-              ss << '/';
-            ss << "realm_" << std::hash<std::string>()(path) << ".note";
-            path = ss.str();
-            ret = mkfifo(path.c_str(), 0600);
-            err = errno;
+
+            std::string named_pipe_dir = Realm::get_named_pipe_directory();
+            if (!named_pipe_dir.empty()) {
+                std::ostringstream ss;
+                ss << named_pipe_dir << "realm_" << std::hash<std::string>()(path) << ".note";
```

Comment:
Using `util::format()` here rather than a stringstream will slightly reduce the size of the compiled binary and make it a bit more concise.